gtk4/src/drawing_area.rs

```diff
@@ -21,4 +28,43 @@ impl<O: IsA<DrawingArea>> DrawingAreaExtManual for O {
             )
         }
     }
+
+    fn set_draw_func<P: FnMut(&DrawingArea, &cairo::Context, i32, i32) + 'static>(
+        &self,
+        draw_func: P,
+    ) {
+        let draw_func_data: Box<RefCell<P>> = Box::new(RefCell::new(draw_func));
+        unsafe extern "C" fn draw_func_func<
+            P: FnMut(&DrawingArea, &cairo::Context, i32, i32) + 'static,
+        >(
+            drawing_area: *mut ffi::GtkDrawingArea,
+            cr: *mut cairo::ffi::cairo_t,
+            width: libc::c_int,
+            height: libc::c_int,
+            user_data: glib::ffi::gpointer,
+        ) {
+            let drawing_area = from_glib_borrow(drawing_area);
+            let cr = from_glib_borrow(cr);
+            let callback: &RefCell<P> = &*(user_data as *mut _);
+            (callback.borrow_mut())(&drawing_area, &cr, width, height);
+        }
+        let draw_func = Some(draw_func_func::<P> as _);
+        unsafe extern "C" fn destroy_func<
+            P: FnMut(&DrawingArea, &cairo::Context, i32, i32) + 'static,
+        >(
+            data: glib::ffi::gpointer,
+        ) {
+            let _callback: Box<RefCell<P>> = Box::from_raw(data as *mut _);
+        }
+        let destroy_call3 = Some(destroy_func::<P> as _);
```

Comment:
Why the `3`?

Reply:
That's from the autogenerated code from gir. I didn't want to change it more than necessary.

FWIW, you named those variables like that :)

Reply:
It's weird to have the count starting at 3. :laughing: 

Well, please clean it up a bit, it's manual code now. :p

Reply:
The reason for keeping it as similar as possible is so that it's easier to autogenerate it again later once we have support for overriding the closure trait